### ISSUE TITLE
eks anywhere 0.8.0 fail to download artifacts 

### ISSUE THREAD

**Issue by dborysenko** [aws/eks-anywhere#1753]
**What happened**:
eksctl anywhere 0.8.0 fails to download artifacts with below error:
```2022-04-04T13:08:40.479-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/infrastructure-docker/v1.0.2/metadata.yaml
2022-04-04T13:08:40.479-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/cluster-api-provider-docker/metadata.yaml
2022-04-04T13:08:40.692-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/infrastructure-docker/v1.0.2/metadata.yaml to eks-anywhere-downloads/1.20/cluster-api-provider-docker/metadata.yaml
2022-04-04T13:08:40.692-0500	V3	Downloading artifact: 
2022-04-04T13:08:40.692-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/cluster-api-provider-cloudstack
Error: error downloading artifact for component cluster-api-provider-cloudstack: failed reading local file [] for cluster spec: open : no such file or directory
```

**What you expected to happen**:
Expected successful artifacts download.

**How to reproduce it (as minimally and precisely as possible)**:
1. Install latest (0.8.0) version of eksctl anywhere tool.
2. Generate sample cluster config:
`eksctl anywhere generate clusterconfig mgmt --provider vsphere > eksa-mgmt-cluster.yaml`

3. Add registryMirrorConfiguration to generated config
```
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: Cluster
metadata:
  name: mgmt
spec:
  clusterNetwork:
    cniConfig:
      cilium: {}
    pods:
      cidrBlocks:
      - 192.168.0.0/16
    services:
      cidrBlocks:
      - 10.96.0.0/12
  controlPlaneConfiguration:
    count: 2
    endpoint:
      host: ""
    machineGroupRef:
      kind: VSphereMachineConfig
      name: mgmt-cp
  datacenterRef:
    kind: VSphereDatacenterConfig
    name: mgmt
  externalEtcdConfiguration:
    count: 3
    machineGroupRef:
      kind: VSphereMachineConfig
      name: mgmt-etcd
  kubernetesVersion: "1.22"
  managementCluster:
    name: mgmt
  workerNodeGroupConfigurations:
  - count: 2
    machineGroupRef:
      kind: VSphereMachineConfig
      name: mgmt
    name: md-0
  registryMirrorConfiguration:
    endpoint: my-docker-registry.example.com
    port: 443

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereDatacenterConfig
metadata:
  name: mgmt
spec:
  datacenter: ""
  insecure: false
  network: ""
  server: ""
  thumbprint: ""

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  name: mgmt-cp
spec:
  datastore: ""
  diskGiB: 25
  folder: ""
  memoryMiB: 8192
  numCPUs: 2
  osFamily: bottlerocket
  resourcePool: ""
  users:
  - name: ec2-user
    sshAuthorizedKeys:
    - ssh-rsa AAAA...

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  name: mgmt
spec:
  datastore: ""
  diskGiB: 25
  folder: ""
  memoryMiB: 8192
  numCPUs: 2
  osFamily: bottlerocket
  resourcePool: ""
  users:
  - name: ec2-user
    sshAuthorizedKeys:
    - ssh-rsa AAAA...

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  name: mgmt-etcd
spec:
  datastore: ""
  diskGiB: 25
  folder: ""
  memoryMiB: 8192
  numCPUs: 2
  osFamily: bottlerocket
  resourcePool: ""
  users:
  - name: ec2-user
    sshAuthorizedKeys:
    - ssh-rsa AAAA...

---
```

4. download artifacts 
`eksctl anywhere download artifacts -f eksa-mgmt-cluster.yaml -v 9 `

5. Download Log:
```
2022-04-04T13:08:36.762-0500	V4	Logger init completed	{"vlevel": 9}
2022-04-04T13:08:36.764-0500	V4	Reading releases manifest	{"url": "https://anywhere-assets.eks.amazonaws.com/releases/eks-a/manifest.yaml"}
2022-04-04T13:08:37.443-0500	V4	Reading bundles manifest	{"url": "https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/manifest.yaml"}
2022-04-04T13:08:38.253-0500	V4	Reading releases manifest	{"url": "https://anywhere-assets.eks.amazonaws.com/releases/eks-a/manifest.yaml"}
2022-04-04T13:08:38.385-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/eks-a/manifest.yaml
2022-04-04T13:08:38.385-0500	V3	Creating local artifact file: eks-anywhere-downloads/manifest.yaml
2022-04-04T13:08:38.462-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/eks-a/manifest.yaml to eks-anywhere-downloads/manifest.yaml
2022-04-04T13:08:38.462-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/etcdadm-controller/manifests/bootstrap-etcdadm-controller/v1.0.0-rc7/bootstrap-components.yaml
2022-04-04T13:08:38.462-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/etcdadm-controller/bootstrap-components.yaml
2022-04-04T13:08:38.556-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/etcdadm-controller/manifests/bootstrap-etcdadm-controller/v1.0.0-rc7/bootstrap-components.yaml to eks-anywhere-downloads/1.20/etcdadm-controller/bootstrap-components.yaml
2022-04-04T13:08:38.556-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/etcdadm-controller/manifests/bootstrap-etcdadm-controller/v1.0.0-rc7/metadata.yaml
2022-04-04T13:08:38.556-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/etcdadm-controller/metadata.yaml
2022-04-04T13:08:38.629-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/etcdadm-controller/manifests/bootstrap-etcdadm-controller/v1.0.0-rc7/metadata.yaml to eks-anywhere-downloads/1.20/etcdadm-controller/metadata.yaml
2022-04-04T13:08:38.629-0500	V3	Downloading artifact: https://distro.eks.amazonaws.com/crds/releases.distro.eks.amazonaws.com-v1alpha1.yaml
2022-04-04T13:08:38.629-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/eks-distro/releases.distro.eks.amazonaws.com-v1alpha1.yaml
2022-04-04T13:08:38.704-0500	V3	Successfully downloaded artifact https://distro.eks.amazonaws.com/crds/releases.distro.eks.amazonaws.com-v1alpha1.yaml to eks-anywhere-downloads/1.20/eks-distro/releases.distro.eks.amazonaws.com-v1alpha1.yaml
2022-04-04T13:08:38.704-0500	V3	Downloading artifact: https://distro.eks.amazonaws.com/kubernetes-1-20/kubernetes-1-20-eks-13.yaml
2022-04-04T13:08:38.704-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/eks-distro/kubernetes-1-20-eks-13.yaml
2022-04-04T13:08:38.797-0500	V3	Successfully downloaded artifact https://distro.eks.amazonaws.com/kubernetes-1-20/kubernetes-1-20-eks-13.yaml to eks-anywhere-downloads/1.20/eks-distro/kubernetes-1-20-eks-13.yaml
2022-04-04T13:08:38.798-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api-provider-aws/manifests/infrastructure-aws/v0.6.4/infrastructure-components.yaml
2022-04-04T13:08:38.798-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/cluster-api-provider-aws/infrastructure-components.yaml
2022-04-04T13:08:39.212-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api-provider-aws/manifests/infrastructure-aws/v0.6.4/infrastructure-components.yaml to eks-anywhere-downloads/1.20/cluster-api-provider-aws/infrastructure-components.yaml
2022-04-04T13:08:39.212-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api-provider-aws/manifests/infrastructure-aws/v0.6.4/cluster-template.yaml
2022-04-04T13:08:39.212-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/cluster-api-provider-aws/cluster-template.yaml
2022-04-04T13:08:39.286-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api-provider-aws/manifests/infrastructure-aws/v0.6.4/cluster-template.yaml to eks-anywhere-downloads/1.20/cluster-api-provider-aws/cluster-template.yaml
2022-04-04T13:08:39.286-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api-provider-aws/manifests/infrastructure-aws/v0.6.4/metadata.yaml
2022-04-04T13:08:39.286-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/cluster-api-provider-aws/metadata.yaml
2022-04-04T13:08:39.426-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api-provider-aws/manifests/infrastructure-aws/v0.6.4/metadata.yaml to eks-anywhere-downloads/1.20/cluster-api-provider-aws/metadata.yaml
2022-04-04T13:08:39.426-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/bootstrap-kubeadm/v1.0.2/bootstrap-components.yaml
2022-04-04T13:08:39.426-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/capi-kubeadm-bootstrap/bootstrap-components.yaml
2022-04-04T13:08:39.713-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/bootstrap-kubeadm/v1.0.2/bootstrap-components.yaml to eks-anywhere-downloads/1.20/capi-kubeadm-bootstrap/bootstrap-components.yaml
2022-04-04T13:08:39.713-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/bootstrap-kubeadm/v1.0.2/metadata.yaml
2022-04-04T13:08:39.713-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/capi-kubeadm-bootstrap/metadata.yaml
2022-04-04T13:08:39.792-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/bootstrap-kubeadm/v1.0.2/metadata.yaml to eks-anywhere-downloads/1.20/capi-kubeadm-bootstrap/metadata.yaml
2022-04-04T13:08:39.792-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/infrastructure-docker/v1.0.2/infrastructure-components-development.yaml
2022-04-04T13:08:39.792-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/cluster-api-provider-docker/infrastructure-components-development.yaml
2022-04-04T13:08:40.095-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/infrastructure-docker/v1.0.2/infrastructure-components-development.yaml to eks-anywhere-downloads/1.20/cluster-api-provider-docker/infrastructure-components-development.yaml
2022-04-04T13:08:40.095-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/infrastructure-docker/v1.0.2/cluster-template-development.yaml
2022-04-04T13:08:40.095-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/cluster-api-provider-docker/cluster-template-development.yaml
2022-04-04T13:08:40.479-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/infrastructure-docker/v1.0.2/cluster-template-development.yaml to eks-anywhere-downloads/1.20/cluster-api-provider-docker/cluster-template-development.yaml
2022-04-04T13:08:40.479-0500	V3	Downloading artifact: https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/infrastructure-docker/v1.0.2/metadata.yaml
2022-04-04T13:08:40.479-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/cluster-api-provider-docker/metadata.yaml
2022-04-04T13:08:40.692-0500	V3	Successfully downloaded artifact https://anywhere-assets.eks.amazonaws.com/releases/bundles/8/artifacts/cluster-api/manifests/infrastructure-docker/v1.0.2/metadata.yaml to eks-anywhere-downloads/1.20/cluster-api-provider-docker/metadata.yaml
2022-04-04T13:08:40.692-0500	V3	Downloading artifact: 
2022-04-04T13:08:40.692-0500	V3	Creating local artifact file: eks-anywhere-downloads/1.20/cluster-api-provider-cloudstack
Error: error downloading artifact for component cluster-api-provider-cloudstack: failed reading local file [] for cluster spec: open : no such file or directory
```

**Anything else we need to know?**:

**Environment**:
- EKS Anywhere Release:
- EKS Distro Release:

**Comment by mrajashree** [comment]
Fixed in release v0.8.1

**Comment by dborysenko** [comment]
@mrajashree I was able to download artifacts using 0.8.1 release.
Thanks.
